proxy/deploy_test.go

```diff
@@ -192,5 +199,23 @@ func Test_DeployFunction_generateFuncStr(t *testing.T) {
 			t.Fatalf("generateFuncStr %s\nwant: %s, got: %s", testCase.name, testCase.expectedStr, funcStr)
 		}
 	}
+}
+
+type TestAuth struct {
```

Comment:
This doesn't need to be public either

Reply:
Actually, i am not even sure you need to save any strings.  If the intent is to provide a mock that we can pass through  for tests, then only thing it needs to do is implement the ClientAuth, I would do something like this

```go
type testAuth struct {
    err error
}

func NewTestAuth(err error) ClientAuth {
    return &testAuth{err: err}
}

func (c *testAuth) Set(req *http.Request) error {
    return c.err
}
```

Which allows us to test how the methods behave when the `Set` fails

Reply:
I liked this and have implemented the same.